os_migrate/plugins/module_utils/server.py

```diff
@@ -248,3 +248,22 @@ def _refs_from_ser(self, conn, filters=None):
         refs['volumes'] = params['volumes']
 
         return refs
+
+    def dst_prerequisites_errors(self, conn, filters=None):
+        """Get messages for unmet destination cloud prerequisites including keypairs.
+
+        Returns: list of strings (error messages), empty when prerequisites
+        are met
+        """
+        errors = super().dst_prerequisites_errors(conn, filters)
+
+        # Parent class dst_prerequisites_errors does not validate keypairs exist in destination cloud.
+        # To do this we do a lookup to see if keypair name exists and append to errors list if not
+        params = self.params()
+        try:
+            conn.compute.find_keypair(params['key_name'], **filters)
```

Comment:
Actually we need to pass `ignore_missing=False`, which the reference lookup methods in the parent class do. Otherwise the method returns None rather than raising an exception.

Reply:
https://docs.openstack.org/openstacksdk/latest/user/proxies/compute.html#keypair-operations

Reply:
https://github.com/os-migrate/os-migrate/blob/d86c0a27fcff1cc208ebe2b244993bfba5cabd22/os_migrate/plugins/module_utils/reference.py#L374-L378

Reply:
Thanks.